packages/transaction-controller/src/TransactionController.ts

```diff
@@ -335,14 +347,12 @@ export class TransactionController extends BaseController<
    * if not provided. If A `<tx.id>:unapproved` hub event will be emitted once added.
    *
    * @param transaction - The transaction object to add.
-   * @param origin - The domain origin to append to the generated TransactionMeta.
-   * @param deviceConfirmedOn - An enum to indicate what device the transaction was confirmed to append to the generated TransactionMeta.
+   * @param opts - The options bag for addTransaction.
    * @returns Object containing a promise resolving to the transaction hash if approved.
    */
   async addTransaction(
     transaction: Transaction,
-    origin?: string,
-    deviceConfirmedOn?: WalletDevice,
+    opts: AddTransactionOptions = {},
```

Comment:
Do we need a separate type here as it won't be used anywhere else? Plus we could inline all the variables and dereference them so we can use them as if they were standard arguments?

As done in the extension for reference.

Reply:
Removed type, and converted to inline variables